pandas/core/arrays/timedeltas.py

```diff
@@ -137,25 +139,67 @@ def dtype(self):
     # Constructors
     _attributes = ["freq"]
 
-    @classmethod
-    def _simple_new(cls, values, freq=None, dtype=_TD_DTYPE):
-        # `dtype` is passed by _shallow_copy in corner cases, should always
-        #  be timedelta64[ns] if present
-        assert dtype == _TD_DTYPE
-        assert isinstance(values, np.ndarray), type(values)
+    def __init__(self, values, dtype=_TD_DTYPE, freq=None, copy=False):
+        if isinstance(values, (ABCSeries, ABCIndexClass)):
+            values = values._values
+
+        if isinstance(values, type(self)):
+            if freq is None:
+                freq = values.freq
+            elif freq and values.freq:
+                freq = to_offset(freq)
+                freq, freq_infer = dtl.validate_inferred_freq(
+                    freq, values.freq,
+                    freq_infer=False
+                )
+            values = values._data
+
+        if not isinstance(values, np.ndarray):
+            msg = (
+                "Unexpected type '{}'. 'values' must be a TimedeltaArray "
```

Comment:
similiar to above, should be an assert or AssertionError

Reply:
Collapsing into 
https://github.com/pandas-dev/pandas/pull/24024/files#r242739777